### PR TITLE
IN-1090 Validation for automated letters

### PR DESCRIPTION
**Pull request by devkokov**
## Purpose

Validate that Sirius sends out automated letters for migrated cases

## Approach

casrec_letters.csv is loaded in alongside main Casrec data. It contains information about automated letters sent from Casrec between 21-25 Feb 2022.

SQL script "generates" the automated letters that Sirius would have sent out on each of those days.

A validation script compares the two tables. 

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have done an adhoc run against preprod (only needed for high complexity PRs)
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: migration_steps/validation/validate_db/app/letters/sirius_dates.json [hunk 1]
```diff
@@ -0,0 +1 @@
...rr1_from":"2022-01-19","rr1_to":"2022-01-19","rr2_from_lay":"2021-12-07","rr2_to_lay":"2021-12-07","rr2_from_pro_pa":"2021-10-26","rr2_to_pro_pa":"2021-10-26","rr3_from":"2022-01-19","rr3_to":"2022-01-19"},"2022-01-24":{"bs1_from":"2022-01-19","bs1_to":"2022-01-19","bs2_from":"2021-12-30","bs2_to":"2021-12-30","rd1_from":"2022-02-14","rd1_to":"2022-02-14","rd2_from":"2022-01-23","rd2_to":"2022-01-23","ri2_from":"2022-01-19","ri2_to":"2022-01-19","ri3_from":"2021-12-30","ri3_to":"2021-12-30","rr1_from":"2022-01-19","rr1_to":"2022-01-19","rr2_from_lay":"2021-12-07","rr2_to_lay":"2021-12-07","rr2_from_pro_pa":"2021-10-26","rr2_to_pro_pa":"2021-10-26","rr3_from":"2022-01-19","rr3_to":"2022-01-19"},"2022-01-25":{"bs1_from":"2022-01-20","bs1_to":"2022-01-20","bs2_from":"2021-12-31","bs2_to":"2021-12-31","rd1_from":"2022-02-15","rd1_to":"2022-02-15","rd2_from":"2022-01-24","rd2_to":"2022-01-24","ri2_from":"2022-01-20","ri2_to":"2022-01-20","ri3_from":"2021-12-31","ri3_to":"2021-12-31","rr1_from":"2022-01-20","rr1_to":"2022-01-20","rr2_from_lay":"2021-12-08","rr2_to_lay":"2021-12-08","rr2_from_pro_pa":"2021-10-27","rr2_to_pro_pa":"2021-10-27","rr3_from":"2022-01-20","rr3_to":"2022-01-20"}}
```
Comment (townxelliot) [inline migration_steps/validation/validate_db/app/letters/sirius_dates.json:1]: What's the purpose of this file, just for my information?

**Review by townxelliot — APPROVED** [review]
I'm not going to pretend I fully understand everything here, but it looks solid to me.

File: migration_steps/validation/validate_db/app/letters/sirius_dates.json (hunk 1 above)
Comment (devkokov) [reply migration_steps/validation/validate_db/app/letters/sirius_dates.json:1]: I wrote a little script in Sirius that takes a date as input and spits out this json of variables (dates) for the letter selection SQL.

For example, if today is `2022-01-21`, the date range for BS1 letters will be from `2022-01-18` to `2022-01-18`.

Basically didn't want to rewrite the date calculation logic in SQL, when I only need to test a few specific dates

**Review by jamesrwarren — APPROVED** [review]
This is also a bit beyond me but I am not one to stand in the way of progress!

**Pull request merged**